pkg/manager/member/pd_scaler.go

```diff
@@ -39,27 +39,42 @@ func NewPDScaler(pdControl pdapi.PDControlInterface,
 	return &pdScaler{generalScaler{pdControl, pvcLister, pvcControl}}
 }
 
+func (psd *pdScaler) Scale(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	if err := validateScaling(oldSet, newSet); err != nil {
+		return err
+	}
+	if *newSet.Spec.Replicas > *oldSet.Spec.Replicas {
+		return psd.ScaleOut(tc, oldSet, newSet)
+	} else if *newSet.Spec.Replicas < *oldSet.Spec.Replicas {
+		return psd.ScaleIn(tc, oldSet, newSet)
+	}
+	return nil
+}
+
 func (psd *pdScaler) ScaleOut(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	ordinal, replicas, deleteSlots, err := scaleOne(oldSet, newSet)
+	if err != nil {
+		return err
+	}
+	resetReplicas(newSet, oldSet)
```

Comment:
I've updated the scaling logic like this:

- get desired replicas and delete slot
- reset the new statefulset
- if our scaling logic succeeds, set desired replicas and delete slots, otherwise, return an error